plugins/jobs/src/js/components/JobsForm.tsx

```diff
@@ -300,19 +303,41 @@ class JobsForm extends React.Component<JobsFormProps, JobsFormState> {
       handleTabChange,
       isJSONModeActive,
       jobSpec,
-      errors
+      errors,
+      i18n
     } = this.props;
+    const errorsByTab = FormErrorUtil.getTopLevelTabErrors(
+      errors,
+      JobErrorTabPathRegexes,
+      ServiceErrorPathMapping,
+      i18n
+    );
     const jobJSON = this.getJSONEditorData(jobSpec);
     const tabList = Hooks.applyFilter(
       "createJobTabList",
       JobsForm.navigationItems
-    ).map((item: NavigationItem) => (
-      <TabButton
-        id={item.id}
-        label={<Trans render="span" id={item.label} />}
-        key={item.key}
-      />
-    ));
+    ).map((item: NavigationItem) => {
+      const finalErrorCount = findNestedPropertyInObject(
+        errorsByTab,
+        `${item.id}.length`
+      );
+
+      return (
+        <TabButton
+          id={item.id}
+          label={<Trans render="span" id={item.label} />}
+          key={item.key}
+          count={finalErrorCount}
+          showErrorBadge={Boolean(finalErrorCount) && this.props.showAllErrors}
+          description={
+            // TODO: pluralize
+            <Trans render="span">
+              {finalErrorCount} issues need addressing
```

Comment:
Can this be `issueCount`, since we already have a translation: https://github.com/dcos/dcos-ui/blob/235e9b80964f96fa7606dad1fb5f4269a33ce0c5/locale/zh/messages.json#L1008